### ISSUE TITLE
Bug: Non-query statements are not taking into account SQL Server table hints

### ISSUE THREAD

**Issue by marcin-brzozowski** [mikependon/RepoDB#533]
The following non-query statements are not taking into account SQL Server table hints:

- Delete
- DeleteAll
- Insert
- InsertAll
- Merge
- MergeAll
- Update
- UpdateAll

The issue is related with the [CommandTextCache](https://github.com/mikependon/RepoDb/blob/master/RepoDb.Core/RepoDb/Cachers/CommandTextCache.cs) class in the `RepoDb.Core` project. The related methods responsible for building the statements don't pass `Hints` object into the statement builders.

**Comment by marcin-brzozowski** [comment]
#534 is created to fix this issue.

**Comment by mikependon** [comment]
The fix is now available at [RepoDb v1.12.0-beta4](https://www.nuget.org/packages/RepoDb/1.12.0-beta4) and [RepoDb.SqlServer v1.1.0-beta2](https://www.nuget.org/packages/RepoDb.SqlServer/1.1.0-beta2).

```csharp
> Install-Package RepoDb -version 1.12.0-beta4
> Install-Package RepoDb.SqlServer -version 1.1.0-beta2
```

**Comment by mikependon** [comment]
This has been fixed and is now available at [RepoDb v1.12.3](https://www.nuget.org/packages/RepoDb/1.12.3). Closing this incident ticket now.
